compiler/src/compile.rs

```diff
@@ -797,11 +811,15 @@ impl<O: OutputStream> Compiler<O> {
         body: &[ast::Statement],
         decorator_list: &[ast::Expression],
         returns: &Option<ast::Expression>, // TODO: use type hint somehow..
+        is_async: bool,
     ) -> Result<(), CompileError> {
         // Create bytecode for this function:
         // remember to restore self.in_loop to the original after the function is compiled
         let was_in_loop = self.in_loop;
         let was_in_function_def = self.in_function_def;
+
+        let was_in_async_func = self.in_async_func;
+        self.in_async_func = is_async;
```

Comment:
Can we be `in_async_func` without `in_function_def` being `true`? Maybe these two flags should be combined to a enum named something like `func_mode`?

Reply:
That would be good, some sort of Context enum